File: Models/NodeData.cs
using System;
using System.Collections.Generic;
using Avalonia;

namespace TeachingAidMac.Models
{
    [Serializable]
    public class NodeData
    {
        public string Name { get; set; } = "";
        public List<ConnectionData> Connections { get; set; } = new List<ConnectionData>();
        public Point LocationOnGraph { get; set; }

        public void SetName(string name)
        {
            Name = name;
        }

        public void AddConnection(ConnectionData connection)
        {
            Connections.Add(connection);
        }

        public void AddLocation(Point location)
        {
            LocationOnGraph = location;
        }

        public string GetName()
        {
            return Name;
        }

        public List<ConnectionData> GetConnections()
        {
            return Connections;
        }

        public Point GetLocation()
        {
            return LocationOnGraph;
        }
    }
}

File: NodeData.vb
<Serializable> 
Public Class NodeData 

    Private Name As String 

    Private Connections As New List(Of ConnectionData) 

    Private LocationOnGraph As Point 

 

    Public Sub SetName(Name As String) 

        Me.Name = Name 

    End Sub 

    Public Sub AddConnection(Connection As ConnectionData) 

        Me.Connections.Add(Connection) 

    End Sub 

    Public Sub AddLocation(MyLocation As Point) 

        Me.LocationOnGraph = MyLocation 

    End Sub 

 

    Public Function GetName() As String 

        Return Me.Name 

    End Function 

    Public Function GetConnections() As List(Of ConnectionData) 

        Return Me.Connections 

    End Function 

    Public Function GetLocation() As Point 

        Return Me.LocationOnGraph 

    End Function 

End Class